wazo_phoned/plugins/blf/services.py

```diff
@@ -0,0 +1,55 @@
+# Copyright 2020 The Wazo Authors  (see the AUTHORS file)
+# SPDX-License-Identifier: GPL-3.0-or-later
+import logging
+
+from xivo import xivo_helpers
+
+logger = logging.getLogger(__name__)
+
+
+class BlfService:
+
+    DEVICE_PATTERN = 'Custom:{}'
+    INUSE = 'INUSE'
+    NOT_INUSE = 'NOT_INUSE'
+
+    def __init__(self, amid_client, confd_client):
+        self.amid = amid_client
+        self.confd = confd_client
+
+    def _device(self, user_uuid, name, destination=''):
+        funckey_prefix = self.confd.extensions_features.list(search='phoneprogfunckey')[
+            'items'
+        ][0]['exten']
+        device_extension = self.confd.extensions_features.list(search=name)['items'][0][
+            'exten'
+        ]
+        user_id = self.confd.users.get(user_uuid)['id']
```

Comment:
I hate doing a GET on /users here.
Since we do not want to keep the `id` field of the user forever I think we should try to find a way to derive a valid extension segment from the UUID or maybe the whole UUID if its possible.
And if we are stuck with the `id` field we should add it to the event to avoid doing this HTTP request

Reply:
I will create another ticket for this one.